allennlp/commands/push_to_hf.py

```diff
@@ -0,0 +1,85 @@
+"""
+The `push_to_hf` subcommand can be used to push a trained model to the
+Hugging Face Hub ([hf.co](https://hf.co/)).
+"""
+
+import argparse
+import logging
+
+from overrides import overrides
+
+from allennlp.commands.subcommand import Subcommand
+from allennlp.common.push_to_hf import push_to_hf
+
+logger = logging.getLogger(__name__)
+
+
+@Subcommand.register("push_to_hf")
+class PushToHf(Subcommand):
+    @overrides
+    def add_subparser(self, parser: argparse._SubParsersAction) -> argparse.ArgumentParser:
+        description = """Push a model to the Hugging Face Hub
+
+        Pushing your models to the Hugging Face Hub ([hf.co](https://hf.co/)) 
+        allows you to share your models with others. On top of that, you can try
+        the models directly in the browser with the available widgets.
+
+        Before running this command, login to Hugging Face with huggingface-cli login
+        """
+        subparser = parser.add_parser(self.name, description=description, help=description)
+        subparser.set_defaults(func=push)
+
+        subparser.add_argument(
+            "-a",
+            "--archive_path",
+            required=True,
+            type=str,
+            help="full path to the zipped model or to a directory with the serialized model.",
+        )
```

Comment:
I don't quite know where to stick it, but it's always better to upload a serialization directory than a model archive, because you get metrics and tensor board, right? Can we capture that somewhere? People need to know!

Reply:
Yes, the archive is mostly intended to easily migrate existing models. WDYT of having two optional params, `serializable-dir` and `archive-path`. Users need to specify one of them and then it's more explicit what we're operating with (and we could even give a warning that serializable dir is recommended)

Reply:
What do you think of autodetecting when a model is inside a serialization dir? Too much magic?

Reply:
I'm ok with either way. I think autodetecting could lead to confusion for the users, but showing a clear warning might be enough

Reply:
Then let's not have auto detection. It's better to be predictable.